apps/daimo-mobile/src/view/screen/send/CreateNoteTab.tsx

```diff
@@ -37,35 +29,20 @@ import { TextCenter, TextError, TextLight } from "../../shared/text";
 
 /** Creates a Note. User picks amount, then sends message via ShareSheet. */
 export function CreateNoteTab({ hide }: { hide: () => void }) {
-  const [ephemeralPrivateKey, setEphemeralPrivateKey] = useState<Hex>();
-
+  const [ephemeralPrivKey] = useState<Hex>(generatePrivateKey);
   const ephemeralOwner = useMemo(
-    () =>
-      ephemeralPrivateKey
-        ? privateKeyToAccount(ephemeralPrivateKey).address
-        : null,
-    [ephemeralPrivateKey]
+    () => ephemeralPrivKey && privateKeyToAccount(ephemeralPrivKey).address,
+    [ephemeralPrivKey]
   );
 
   const [dollars, setDollars] = useState(0);
 
   const [account] = useAccount();
   assert(account != null);
-  const { address } = account;
-
-  const [isNotesContractApproved, setIsNotesContractApproved] = useState(false);
-
-  useEffect(() => {
-    (async () => {
-      // TODO: pull this logic out of the UI.
-      // + automatically approve Notes contract on account creation
-      const allowance = await fetchNotesContractAllowance(address);
-      setIsNotesContractApproved(allowance > 0);
 
-      const privKey = generatePrivateKey();
-      setEphemeralPrivateKey(privKey);
-    })();
-  }, []);
+  // TODO: load and cache
+  // Minor optimization. Approving unconditionally costs a bit more gas.
+  const approveFirst = true;
```

Comment:
I think we should consider moving this approve to the onboarding deployment step: would make `@daimo/userop` slightly cleaner as well